crates/bevy_ecs/src/storage/table/mod.rs

```diff
@@ -682,29 +391,115 @@ impl Table {
         }
     }
 
-    /// Fetches a read-only reference to the [`Column`] for a given [`Component`] within the
-    /// table.
+    /// Get the data of the column matching `component_id` as a slice.
+    ///
+    /// # Safety
+    /// `row.as_usize()` < `self.len()`
+    /// - `T` must match the `component_id`
+    pub unsafe fn get_data_slice_for<T>(
+        &self,
+        component_id: ComponentId,
+    ) -> Option<&[UnsafeCell<T>]> {
+        self.get_column(component_id)
+            .map(|col| col.data.get_sub_slice(self.len()))
+    }
+
+    /// Get the added ticks of the column matching `component_id` as a slice.
+    ///
+    /// # Safety
+    /// `row.as_usize()` < `self.len()`
+    /// - `T` must match the `component_id`
+    pub unsafe fn get_added_ticks_slice_for(
+        &self,
+        component_id: ComponentId,
+    ) -> Option<&[UnsafeCell<Tick>]> {
+        self.get_column(component_id)
+            .map(|col| col.added_ticks.as_slice(self.len()))
+    }
+
+    /// Get the changed ticks of the column matching `component_id` as a slice.
+    ///
+    /// # Safety
+    /// `row.as_usize()` < `self.len()`
+    /// - `T` must match the `component_id`
+    pub unsafe fn get_changed_ticks_slice_for(
+        &self,
+        component_id: ComponentId,
+    ) -> Option<&[UnsafeCell<Tick>]> {
+        self.get_column(component_id)
+            .map(|col| col.changed_ticks.as_slice(self.len()))
+    }
+
+    /// Get the specific [`change tick`](Tick) of the component matching `component_id` in `row`.
+    ///
+    /// # Safety
+    /// - `row.as_usize()` < `self.len()`
+    /// - `T` must match the `component_id`
+    pub unsafe fn get_column_changed_tick(
+        &self,
+        component_id: ComponentId,
+        row: TableRow,
+    ) -> &UnsafeCell<Tick> {
+        self.get_column(component_id)
+            .debug_checked_unwrap()
+            .changed_ticks
+            .get_unchecked(row.as_usize())
+    }
+
+    /// Get the specific [`add tick`](Tick) of the component matching `component_id` in `row`.
+    ///
+    /// # Safety
+    /// `row.as_usize()` < `self.len()`
+    /// - `T` must match the `component_id`
+    pub unsafe fn get_column_added_tick(
+        &self,
+        component_id: ComponentId,
+        row: TableRow,
+    ) -> &UnsafeCell<Tick> {
+        self.get_column(component_id)
+            .debug_checked_unwrap()
```

Comment:
Why is this unchecked while the slice ones unsafe? Shouldn't we just handle the failure cases properly, and then use `debug_checked_unwrap` on the output if that fails? This could be a safe function. Same with below.